### ISSUE TITLE
Don't perform a long poll of the execution result if nobody waits on it

### ISSUE THREAD

**Issue by mfateev** [temporalio/sdk-java#343]
Currently WorkflowClient.execute returns CompletableFuture. The problem is that the GetWorkflowExecutionHistory is initiated as soon as the execute is called, not when future.get is called. So if future.get is not ever called the long poll GetWorkflowExecutionHistory is still done.

**Comment by Spikhalskiy** [comment]
Related (duplicate?) to #1212

**Comment by Quinn-With-Two-Ns** [comment]
Closing as duplicate of https://github.com/temporalio/sdk-java/issues/1212
